documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+
+`Keycloak Authorization Services` use several concepts that together take part in defining, and applying access control to resources.
+
+`Resources` define _what_ we are protecting from unauthorized access.
+Each resource can contain a list of `authorization scopes` - actions that are available on the resource, so that permission on a resource can be granted for one or more actions only.
+`Policies` define the groups of users we want to target with permissions. Users can be targeted based on group membership, assigned roles, or individually.
+Finally, the `permissions` tie together specific `resources`, `action scopes` and `policies` to define that 'specific users U can perform certain actions A on the resource R'.
+
+You can read more about `Keycloak Authorization Services` on link:https://www.keycloak.org/docs/latest/authorization_services/index.html[project's web site], but note that the documentation there is focused primarily on using the Authorization Services to protect web resources, whereas here we're using it to protect access to Kafka.
+
+If we take a look under the `Resources` sub-tab of `Authorization` tab, we'll see the list of resource definitions.
+These are resource specifiers - patterns in a specific format, that are used to target policies to specific resources.
+The format is quite simple. For example:
+
+- `kafka-cluster:cluster-1,Topic:a_*`  ... targets only topics in kafka cluster 'cluster-1' with names starting with 'a_'
+
+If `kafka-cluster:XXX` segment is not present, the specifier targets any cluster.
+
+- `Group:x_*` ... targets all consumer groups on any cluster with names starting with 'x_'
+
+The possible resource types mirror the xref:kafka_authorization_model[Kafka authorization model] (`Topic`, `Group`, `Cluster`, ...).
+
+Under `Authorization Scopes` we can see a list of all the possible actions (Kafka permissions) that can be granted on resources of different types.
+It requires some understanding of link:https://kafka.apache.org/documentation/#resources_in_kafka[Kafka's permissions model] to know which of these make sense with which resource type.
+This list mirrors Kafka permissions and should be the same for any deployment.
+
+There is an link:https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/master/oauth-keycloak-authorizer/etc/authorization-scopes.json[authorization-scopes.json] file containing the authorization scopes that can be imported, so that they do not have to be manually entered for every new `Authorization Services` enabled client.
+In order to import `authorization-scopes.json` into a new client, first make sure the new client is `Authorization Enabled` and saved. Then, click on the `Authorization` tab and use the `Import` to import the file. Afterwards, if you select the `Authorization Scopes` you will see the loaded scopes.
+For this example the authorization scopes have already been imported as part of the realm import.
```

Comment:
```suggestion
Rather than importing authorization scopes manually into every new client, you can use a link:https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/master/oauth-keycloak-authorizer/etc/authorization-scopes.json[authorization-scopes.json] file.

You must make sure that the client is enabled in Keycloak for Authorization Services. After which, you can import a JSON file from the *Authorization* tab. You can see the loaded scopes from the *Authorization Scopes* page.

For this example the authorization scopes have already been imported as part of the realm import.
```